mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,526 @@
+                    spc[pt_i, deg ** 2 + deg + order - 1] = int_pts_sum
+
+        # Scale magnetometers and normalize basis vectors to unity magnitude
+        spc *= coil_scale[:, np.newaxis]
+        spc /= np.sqrt(np.sum(spc * spc, axis=0))[np.newaxis, :]
+
+    return S_in, S_out
+
+
+def _alegendre_deriv(degree, order, val):
+    """Compute the derivative of the associated Legendre polynomial at a value.
+
+    Parameters
+    ----------
+    degree : int
+        Degree of spherical harmonic. (Usually) corresponds to 'l'
+    order : int
+        Order of spherical harmonic. (Usually) corresponds to 'm'
+    val : float
+        Value to evaluate the derivative at
+
+    Returns
+    -------
+    dPlm : float
+        Associated Legendre function derivative
+    """
+    from scipy.special import lpmv
+
+    C = 1
+    if order < 0:
+        order = abs(order)
+        C = (-1) ** order * factorial(degree - order) / factorial(degree +
+                                                                  order)
+    return C * (order * val * lpmv(order, degree, val) + (degree + order) *
+                (degree - order + 1) * np.sqrt(1 - val ** 2) *
+                lpmv(order - 1, degree, val)) / (1 - val ** 2)
+
+
+def _grad_in_components(degree, order, rad, az, pol):
+    """Compute gradient of internal component of V(r) spherical expansion.
+
+    Internal component has form: Ylm(pol, az) / (rad ** (degree + 1))
+
+    Parameters
+    ----------
+    degree : int
+        Degree of spherical harmonic. (Usually) corresponds to 'l'
+    order : int
+        Order of spherical harmonic. (Usually) corresponds to 'm'
+    rad : ndarray, shape (n_samples,)
+        Array of radii
+    az : ndarray, shape (n_samples,)
+        Array of azimuthal (longitudinal) spherical coordinates [0, 2*pi]. 0 is
+        aligned with x-axis.
+    pol : ndarray, shape (n_samples,)
+        Array of polar (or colatitudinal) spherical coordinates [0, pi]. 0 is
+        aligned with z-axis.
+
+    Returns
+    -------
+    grads : ndarray, shape (n_samples, 3)
+        Gradient of the spherical harmonic and vector specified in rectangular
+        coordinates
+    """
+    from scipy.special import sph_harm
+    # Compute gradients for all spherical coordinates (Eq. 6)
+    g_rad = -(degree + 1) / rad ** (degree + 2) * sph_harm(order, degree,
+                                                           az, pol)
+
+    g_az = 1 / (rad ** (degree + 2) * np.sin(pol)) * 1j * order * \
+        sph_harm(order, degree, az, pol)
+
+    g_pol = 1 / rad ** (degree + 2) * np.sqrt((2 * degree + 1) *
+                                              factorial(degree - order) /
+                                              (4 * np.pi *
+                                               factorial(degree + order))) * \
+        -np.sin(pol) * _alegendre_deriv(degree, order, np.cos(pol)) * \
+        np.exp(1j * order * az)
```

Comment:
you can write this in a pep8 compliant manner without stacking the code at the right corner, just as we dicussed offline.